asteroid/engine/system.py

```diff
@@ -163,23 +163,23 @@ def validation_epoch_end(self, outputs):
 
     def configure_optimizers(self):
         """Initialize optimizers, batch-wise and epoch-wise schedulers."""
-
-        if self.scheduler is not None:
-            if not isinstance(self.scheduler, (list, tuple)):
-                self.scheduler = [self.scheduler]  # support multiple schedulers
-            epoch_schedulers = []
-            for sched in self.scheduler:
-                if not isinstance(sched, dict):
-                    epoch_schedulers.append(sched)
-                else:
-                    assert sched["interval"] in [
-                        "batch",
-                        "epoch",
-                    ], "Scheduler interval should be either batch or epoch"
-                    if sched["interval"] == "epoch":
-                        epoch_schedulers.append(sched)
-            return [self.optimizer], epoch_schedulers
-        return self.optimizer
+        if self.scheduler is None:
+            return self.optimizer
+
+        if not isinstance(self.scheduler, (list, tuple)):
+            self.scheduler = [self.scheduler]  # support multiple schedulers
+
+        epoch_schedulers = []
+        for sched in self.scheduler:
+            if not isinstance(sched, dict):
+                epoch_schedulers.append(sched)
+            else:
+                assert sched["interval"] in [
+                    "batch",
+                    "epoch",
+                ], "Scheduler interval should be either batch or epoch"
+                epoch_schedulers.append(sched)
```

Comment:
Sorry, the diff is not easily readable..
Only thing which was changed: sched is added to `epoch_schedulers` whatever happens (so that pl handles the saving).